### PR TITLE
Improve support for JMS dictionaries (associative arrays)

### PR DESCRIPTION
**Pull request by goetas**
Doc: https://swagger.io/docs/specification/data-models/dictionaries/

### CODE REVIEW

File: ModelDescriber/JMSModelDescriber.php [hunk 1]
```diff
@@ -92,33 +92,35 @@ public function describe(Model $model, Schema $schema)
...
-                $property->setType('array');
-                $property = $property->getItems();
+            if ($nestedType = $this->getNestedTypeInArray($item)) {
+                list($type, $isHash) = $nestedType;
+                if ($isHash) {
+                    $property->setType('object');
+
+                    $typeDef = $this->findPropertyType($type, $groups);
+                    $ref = new \ReflectionObject($property);
+                    $p = $ref->getProperty('additionalProperties');
```
Comment (goetas) [inline ModelDescriber/JMSModelDescriber.php]: Not really happy about this but the current swagger-php api does not allow to set `additionalProperties` except via constructor args

File: ModelDescriber/JMSModelDescriber.php [hunk 2]
```diff
@@ -92,33 +92,35 @@ public function describe(Model $model, Schema $schema)
                 continue;
             }
 
-            if ($type = $this->getNestedTypeInArray($item)) {
-                $property->setType('array');
-                $property = $property->getItems();
+            if ($nestedType = $this->getNestedTypeInArray($item)) {
+                list($type, $isHash) = $nestedType;
```
Comment (lsmith77) [inline ModelDescriber/JMSModelDescriber.php]: I guess unless we introduce a new class for the type, we are stuck having to build an array as the return value :-/

Comment (goetas) [reply ModelDescriber/JMSModelDescriber.php]: true.. but not sure if it makes sense to introduce a new class a private method

File: ModelDescriber/JMSModelDescriber.php (hunk 1 above)
Comment (GuilhemN) [reply ModelDescriber/JMSModelDescriber.php]: Actually you can call `$schema->merge(['additionnalProperties' => ...])`. (The absence of a setter is most likely just a mistake)

Comment (goetas) [reply ModelDescriber/JMSModelDescriber.php]: ah! nice!

Comment (goetas) [reply ModelDescriber/JMSModelDescriber.php]: fixed

File: ModelDescriber/JMSModelDescriber.php (hunk 2 above)
Comment (GuilhemN) [reply ModelDescriber/JMSModelDescriber.php]: I think it's fine to use an array in this case

Comment (lsmith77) [reply ModelDescriber/JMSModelDescriber.php]: yeah. if we dont have a general case where it would help and given that its a private method

**Comment by GuilhemN** [comment]
Thank you @goetas :)

**Pull request merged**